File: SjoslagetFrontend/lib/model/deleted_booking.dart
import 'package:decimal/decimal.dart';

import 'json_field.dart';

class DeletedBooking {
	final String reference;
	final String firstName;
	final String lastName;
	final String phoneNo;
	final String email;
	final Decimal totalPrice;
	final Decimal amountPaid;
	final DateTime updated;
	final DateTime deleted;

	DeletedBooking(this.reference, this.firstName, this.lastName, this.phoneNo, this.email, this.totalPrice, this.amountPaid, this.updated, this.deleted);

	factory DeletedBooking.fromMap(Map<String, dynamic> map) =>
		DeletedBooking(
			map[REFERENCE],
			map[FIRSTNAME],
			map[LASTNAME],
			map[PHONE_NO],
			map[EMAIL],
			Decimal.parse(map[TOTAL_PRICE].toString()),
			Decimal.parse(map[AMOUNT_PAID].toString()),
			DateTime.parse(map[UPDATED]),
			DateTime.parse(map[DELETED])
		);
}

File: SjoslagetBackend/Models/DeletedBooking.cs
using System;

namespace Accidis.Sjoslaget.WebService.Models
{
	public sealed class DeletedBooking
	{
		public Guid Id { get; set; }
		public Guid CruiseId { get; set; }
		public string Reference { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string Email { get; set; }
		public string PhoneNo { get; set; }
		public decimal TotalPrice { get; set; }
		public decimal AmountPaid { get; set; }
		public DateTime Created { get; set; }
		public DateTime Updated { get; set; }
		public DateTime Deleted { get; set; }

		public static DeletedBooking FromBooking(Booking booking, decimal amountPaid)
		{
			return new DeletedBooking
			{
				CruiseId = booking.CruiseId,
				Reference = booking.Reference,
				FirstName = booking.FirstName,
				LastName = booking.LastName,
				Email = booking.Email,
				PhoneNo = booking.PhoneNo,
				TotalPrice = booking.TotalPrice,
				AmountPaid = amountPaid,
				Created = booking.Created,
				Updated = booking.Updated
			};
		}
	}
}
